import{_ as g,c as i,j as e,a,b as l,w as o,a6 as d,a5 as r,B as c,o as n,G as m}from"./chunks/framework.CHhwEXkI.js";const H=JSON.parse('{"title":"Bioinfo-Overview","description":"","frontmatter":{"date":"2024-10-30T00:00:00.000Z","title":"Bioinfo-Overview","status":"DONE","author":["AllenYGY"],"tags":["NOTE","Review","Bioinfo"],"publish":true},"headers":[],"relativePath":"posts/Bioinfo/Bioinfo-Midterm-Review.md","filePath":"posts/Bioinfo/Bioinfo-Midterm-Review.md","lastUpdated":null}'),p={name:"posts/Bioinfo/Bioinfo-Midterm-Review.md"},h={tabindex:"0",class:"MathJax",jax:"SVG",display:"true",style:{direction:"ltr",display:"block","text-align":"center",margin:"1em 0",position:"relative"}},u={style:{overflow:"visible","min-height":"1px","min-width":"1px","vertical-align":"-1.552ex"},xmlns:"http://www.w3.org/2000/svg",width:"18.789ex",height:"5.036ex",role:"img",focusable:"false",viewBox:"0 -1540 8304.6 2226","aria-hidden":"true"},Q={class:"MathJax",jax:"SVG",style:{direction:"ltr",position:"relative"}},T={style:{overflow:"visible","min-height":"1px","min-width":"1px","vertical-align":"-0.05ex"},xmlns:"http://www.w3.org/2000/svg",width:"1.719ex",height:"1.645ex",role:"img",focusable:"false",viewBox:"0 -705 760 727","aria-hidden":"true"},f={class:"MathJax",jax:"SVG",style:{direction:"ltr",position:"relative"}},b={style:{overflow:"visible","min-height":"1px","min-width":"1px","vertical-align":"0"},xmlns:"http://www.w3.org/2000/svg",width:"2.009ex",height:"1.545ex",role:"img",focusable:"false",viewBox:"0 -683 888 683","aria-hidden":"true"},A={class:"MathJax",jax:"SVG",style:{direction:"ltr",position:"relative"}},x={style:{overflow:"visible","min-height":"1px","min-width":"1px","vertical-align":"0"},xmlns:"http://www.w3.org/2000/svg",width:"1.541ex",height:"1.545ex",role:"img",focusable:"false",viewBox:"0 -683 681 683","aria-hidden":"true"};function w(y,t,q,v,k,S){const s=c("Mermaid");return n(),i("div",null,[t[13]||(t[13]=e("h1",{id:"bioinformatics-review",tabindex:"-1"},[a("Bioinformatics-Review "),e("a",{class:"header-anchor",href:"#bioinformatics-review","aria-label":'Permalink to "Bioinformatics-Review"'},"​")],-1)),t[14]||(t[14]=e("hr",null,null,-1)),t[15]||(t[15]=e("p",null,[e("em",null,"Bioinformatics"),a(": an interdisciplinary ﬁeld that develops and applies computer and computational technologies to study biomedical questions")],-1)),t[16]||(t[16]=e("p",null,"The -informatics in Bioinformatics",-1)),(n(),l(d,null,{default:o(()=>[m(s,{id:"mermaid-10",class:"mermaid",graph:"flowchart%20LR%0A%20%20%20%20D%5BModeling%2FSimulation%5D%0A%20%20%20%20C%5BData%20Mining%5D%0A%20%20%20%20B%5BData%20Computation%5D%0A%20%20%20%20A%5BData%20Management%5D%0A%0A%20%20%20%20subgraph%20Step4%0A%20%20%20%20%20%20%20%20direction%20BT%0A%20%20%20%20%20%20%20%20D4%5BPredictive%20models%5CnSystems%20simulation%5D%0A%20%20%20%20%20%20%20%20D4%20--%3E%20D%0A%20%20%20%20end%0A%0A%20%20%20%20subgraph%20Step3%0A%20%20%20%20%20%20%20%20direction%20BT%0A%20%20%20%20%20%20%20%20D3%5BBiological%20discoveries%5D%0A%20%20%20%20%20%20%20%20D3%20--%3E%20C%0A%20%20%20%20end%0A%0A%20%20%20%20subgraph%20Step2%0A%20%20%20%20%20%20%20%20direction%20BT%0A%20%20%20%20%20%20%20%20D2%5BAlgorithms%5CnSoftware%20tools%5CnWeb%20servers%5D%0A%20%20%20%20%20%20%20%20D2%20--%3E%20B%0A%20%20%20%20end%0A%0A%20%20%20%20subgraph%20Step1%0A%20%20%20%20%20%20%20%20direction%20BT%0A%20%20%20%20%20%20%20%20D1%5BDatabases%5CnOntologies%5CnMeta-data%5D%0A%20%20%20%20%20%20%20%20D1%20--%3E%20A%0A%20%20%20%20end%0A%0A%20%20%20%20%25%25%20%E5%88%9B%E5%BB%BA%E9%9A%90%E8%97%8F%E8%8A%82%E7%82%B9%E4%BB%A5%E8%BF%9E%E6%8E%A5%E5%AD%90%E5%9B%BE%0A%0A%20%20%20%20Step1%20--%3E%20Step2%0A%20%20%20%20Step2%20--%3E%20Step3%0A%20%20%20%20Step3%20--%3E%20Step4%0A%0A%0A%0A"})]),fallback:o(()=>t[0]||(t[0]=[a(" Loading... ")])),_:1})),t[17]||(t[17]=r('<h2 id="chapter-2-part1-gene-and-rna" tabindex="-1">Chapter 2 Part1- Gene and RNA <a class="header-anchor" href="#chapter-2-part1-gene-and-rna" aria-label="Permalink to &quot;Chapter 2 Part1- Gene and RNA&quot;">​</a></h2><div class="info custom-block github-alert"><p class="custom-block-title">+ Gene</p><p>A gene is a locus (or region) of DNA that encodes a functional protein or RNA product, and is the molecular unit of heredity.</p></div><h3 id="central-dogma-and-splicing" tabindex="-1">Central dogma and splicing <a class="header-anchor" href="#central-dogma-and-splicing" aria-label="Permalink to &quot;Central dogma and splicing&quot;">​</a></h3><p><img src="https://cdn.jsdelivr.net/gh/AllenYGY/ImageSpace@main/uPic/t0jEFX.png" alt="Central dogma"></p><h3 id="gene-finding-approaches" tabindex="-1">Gene Finding Approaches <a class="header-anchor" href="#gene-finding-approaches" aria-label="Permalink to &quot;Gene Finding Approaches&quot;">​</a></h3><ul><li>Something that matches <em>statistical patterns</em> common to all genes</li><li>Something that matches an <em>already known gene</em> (homology)</li><li>Hybrid</li></ul><h3 id="gene-measurement" tabindex="-1">Gene Measurement <a class="header-anchor" href="#gene-measurement" aria-label="Permalink to &quot;Gene Measurement&quot;">​</a></h3><ul><li><p>ORF (<em>Open Reading Frame</em>)</p><ul><li>Start codon <em>ATG</em></li><li>Stop codon <em>TAA</em>, <em>TAG</em>, <em>TGA</em></li></ul></li><li><p>Codon Usage</p><ul><li>The preference for certain synonymous codons, often measured by the Codon Adaptation Index (CAI), which reflects the efficiency of gene expression in an organism.</li></ul></li><li><p>Features and motifs</p><ul><li>Promoters, splice sites, enhancers, untranslated regions (UTRs)</li></ul></li></ul><h3 id="gene-prediction" tabindex="-1">Gene Prediction <a class="header-anchor" href="#gene-prediction" aria-label="Permalink to &quot;Gene Prediction&quot;">​</a></h3><p><strong>Similarity-Based Approach to gene prediction</strong></p><ul><li>Genes in different organisms are similar</li></ul><p>The similarity-based approach uses known genes in one genome to predict (unknown) genes in another genome</p><p><strong>Problem</strong> Given a <em>known</em> gene and an <em>unannotated</em> genome sequence, find <em>a set of substrings</em> of the genome sequence that best fits the known gene.</p><h4 id="exon-chaining-algorithm" tabindex="-1">Exon Chaining Algorithm <a class="header-anchor" href="#exon-chaining-algorithm" aria-label="Permalink to &quot;Exon Chaining Algorithm&quot;">​</a></h4><h3 id="non-coding-rna-ncrna" tabindex="-1">Non-coding RNA <em>ncRNA</em> <a class="header-anchor" href="#non-coding-rna-ncrna" aria-label="Permalink to &quot;Non-coding RNA *ncRNA*&quot;">​</a></h3><h4 id="transfer-rnas-trnas" tabindex="-1">Transfer RNAs (tRNAs) <a class="header-anchor" href="#transfer-rnas-trnas" aria-label="Permalink to &quot;Transfer RNAs (tRNAs)&quot;">​</a></h4><ul><li>the first identified RNA class.</li><li>Function as <em>intermediaries between DNA and amino acids in protein synthesis.</em></li><li>Approximately 80 nucleotides (nt) in length.</li><li>Have a cloverleaf-like secondary structure.</li><li>Four short double-helical elements.</li><li>Three loops: D loop, anticodon loop, and T loop.</li></ul><h4 id="micrornas-mirnas" tabindex="-1">microRNAs (miRNAs) <a class="header-anchor" href="#micrornas-mirnas" aria-label="Permalink to &quot;microRNAs (miRNAs)&quot;">​</a></h4><ul><li>Typically originates from non-protein-coding genomic regions or introns.</li><li><strong>Regulates gene expression</strong> by binding to target mRNAs, suppressing translation or promoting mRNA degradation.</li><li><strong>Plays a role in development and differentiation</strong> by controlling cell growth, differentiation, and apoptosis.</li><li><strong>Involved in disease mechanisms</strong>; miRNA dysregulation is linked to cancers, cardiovascular diseases, and neurological disorders.</li></ul><h2 id="chapter-2-part2-genome-ngs-transcriptome-and-assembly" tabindex="-1">Chapter 2 Part2- Genome, NGS, Transcriptome and Assembly <a class="header-anchor" href="#chapter-2-part2-genome-ngs-transcriptome-and-assembly" aria-label="Permalink to &quot;Chapter 2 Part2- Genome, NGS, Transcriptome and Assembly&quot;">​</a></h2><h3 id="snp-single-nucleotide-polymorphism" tabindex="-1">SNP (Single Nucleotide Polymorphism) <a class="header-anchor" href="#snp-single-nucleotide-polymorphism" aria-label="Permalink to &quot;SNP (Single Nucleotide Polymorphism)&quot;">​</a></h3><ul><li>Mutation of a single nucleotide (ACTG)</li><li>Some can be associated with various phenotypic differences</li><li>SNPs are the most common type of genetic variation among people</li></ul><h3 id="cnv-copy-number-variation" tabindex="-1">CNV (Copy Number Variation) <a class="header-anchor" href="#cnv-copy-number-variation" aria-label="Permalink to &quot;CNV (Copy Number Variation)&quot;">​</a></h3><ul><li><strong>Copy Number Variation (CNV)</strong>: Refers to large segments of DNA, typically over 1 kilobase (Kb) in length, that vary in copy number compared to a reference genome.</li><li><strong>Types of CNVs</strong>: Includes deletions, duplications, and insertions.</li><li><strong>Importance of Studying CNVs</strong>: <ul><li>Can impact gene expression and adaptation.</li><li>Provides insight into the complexity of phenotypic variation and disease mechanisms.</li></ul></li></ul><h3 id="gwas-genome-wide-association-studies" tabindex="-1">GWAS - Genome-Wide Association Studies <a class="header-anchor" href="#gwas-genome-wide-association-studies" aria-label="Permalink to &quot;GWAS - Genome-Wide Association Studies&quot;">​</a></h3><ul><li><strong>Genome Wide Association Study (GWAS)</strong>: A study of genetic variations across the genome to associate single nucleotide polymorphisms (SNPs) with traits or disease conditions.</li><li><strong>Purpose</strong>: <ul><li>Enhances understanding of biological processes affecting health.</li><li>Improves disease prediction and patient care.</li><li>Supports the advancement of personalized medicine.</li></ul></li></ul><h3 id="read-a-short-dna-fragment-that-is-sequenced-by-a-sequencer" tabindex="-1">Read: a short DNA fragment that is sequenced by a sequencer. <a class="header-anchor" href="#read-a-short-dna-fragment-that-is-sequenced-by-a-sequencer" aria-label="Permalink to &quot;Read: a short DNA fragment that is sequenced by a sequencer.&quot;">​</a></h3><ul><li><strong>Contents</strong><ul><li><strong>DNA sequence</strong> (represented by symbols).</li><li><strong>Quality information</strong> indicating the sequencing quality of each base.</li></ul></li><li><strong>FASTQ format</strong>: A commonly used file format for storing sequenced DNA fragments along with their quality scores. Each record includes a sequence ID, DNA sequence, separator, and quality scores.</li></ul><h3 id="rna-seq" tabindex="-1">RNA-Seq <a class="header-anchor" href="#rna-seq" aria-label="Permalink to &quot;RNA-Seq&quot;">​</a></h3><p>This flowchart follows the sequence:</p><ol><li>Start with the RNA sample.</li><li>Convert RNA to cDNA using reverse transcription.</li><li>Amplify the cDNA using PCR.</li><li>Fragment the amplified cDNA.</li><li>Sequence the fragments to produce reads.</li></ol>',31)),(n(),l(d,null,{default:o(()=>[m(s,{id:"mermaid-323",class:"mermaid",graph:"flowchart%20TD%0A%20%20%20%20A%5BSample%20RNA%5D%20--%3E%20B%5BReverse%20Transcription%5D%0A%20%20%20%20B%20--%3E%20C%5BPCR%20Amplification%5D%0A%20%20%20%20C%20--%3E%20D%5BAmplified%20cDNA%5D%0A%20%20%20%20D%20--%3E%20E%5BFragmentation%5D%0A%20%20%20%20E%20--%3E%20F%5BSequencing%20Machine%5D%0A%20%20%20%20F%20--%3E%20G%5BReads-Short%20DNA%20Sequences%5D%0A%0A%20%20%20%20%25%25%20Add%20description%20for%20each%20stage%0A%20%20%20%20B%20--%20Convert%20RNA%20to%20cDNA%20--%3E%20C%0A%20%20%20%20C%20--%20Create%20more%20cDNA%20copies%20--%3E%20D%0A%20%20%20%20E%20--%20Break%20into%20smaller%20pieces%20--%3E%20F%0A"})]),fallback:o(()=>t[1]||(t[1]=[a(" Loading... ")])),_:1})),t[18]||(t[18]=r('<h3 id="transcript-abundance" tabindex="-1">Transcript Abundance <a class="header-anchor" href="#transcript-abundance" aria-label="Permalink to &quot;Transcript Abundance&quot;">​</a></h3><ul><li><strong>Definition</strong>: Transcript abundance refers to the amount of mRNA molecules for a specific gene in a sample, indicating the gene&#39;s expression level.</li><li><strong>Importance</strong>: Higher transcript abundance reflects greater gene activity in the cell and often corresponds to genes critical for cell function.</li><li><strong>In RNA Sequencing</strong>: Genes with higher transcript abundance produce more reads in RNA sequencing, as they are more prevalent in the sample.</li></ul><h3 id="sequencing-depth" tabindex="-1">Sequencing Depth <a class="header-anchor" href="#sequencing-depth" aria-label="Permalink to &quot;Sequencing Depth&quot;">​</a></h3><ul><li><strong>Definition</strong>: Sequencing depth is the total number of reads generated in a sequencing experiment, representing the coverage level of RNA in the sample.</li><li><strong>Importance</strong>: Higher sequencing depth increases sensitivity to detect low-abundance genes and improves accuracy in measuring gene expression.</li><li><strong>Normalization</strong>: When comparing gene expression across experiments, sequencing depth needs to be normalized to ensure accurate and fair comparisons of expression levels.</li></ul><blockquote><p>[!abstract]+</p><ul><li><strong>Transcript Abundance</strong> affects the number of reads a gene produces; higher abundance results in more reads.</li><li><strong>Sequencing Depth</strong> determines the coverage and accuracy of RNA sequencing data, with higher depth allowing better detection of low-abundance genes and requiring normalization for cross-experiment comparisons.</li></ul></blockquote><h3 id="expression-level" tabindex="-1">Expression Level <a class="header-anchor" href="#expression-level" aria-label="Permalink to &quot;Expression Level&quot;">​</a></h3><p><strong>RPKM</strong> <strong>(Reads Per Kilobase Million)</strong>: The formula for calculating RPKM (Reads Per Kilobase of transcript, per Million mapped reads) is:</p>',7)),e("mjx-container",h,[(n(),i("svg",u,t[2]||(t[2]=[r('<g stroke="currentColor" fill="currentColor" stroke-width="0" transform="scale(1,-1)"><g data-mml-node="math"><g data-mml-node="mtext"><path data-c="52" d="M130 622Q123 629 119 631T103 634T60 637H27V683H202H236H300Q376 683 417 677T500 648Q595 600 609 517Q610 512 610 501Q610 468 594 439T556 392T511 361T472 343L456 338Q459 335 467 332Q497 316 516 298T545 254T559 211T568 155T578 94Q588 46 602 31T640 16H645Q660 16 674 32T692 87Q692 98 696 101T712 105T728 103T732 90Q732 59 716 27T672 -16Q656 -22 630 -22Q481 -16 458 90Q456 101 456 163T449 246Q430 304 373 320L363 322L297 323H231V192L232 61Q238 51 249 49T301 46H334V0H323Q302 3 181 3Q59 3 38 0H27V46H60Q102 47 111 49T130 61V622ZM491 499V509Q491 527 490 539T481 570T462 601T424 623T362 636Q360 636 340 636T304 637H283Q238 637 234 628Q231 624 231 492V360H289Q390 360 434 378T489 456Q491 467 491 499Z" style="stroke-width:3;"></path><path data-c="50" d="M130 622Q123 629 119 631T103 634T60 637H27V683H214Q237 683 276 683T331 684Q419 684 471 671T567 616Q624 563 624 489Q624 421 573 372T451 307Q429 302 328 301H234V181Q234 62 237 58Q245 47 304 46H337V0H326Q305 3 182 3Q47 3 38 0H27V46H60Q102 47 111 49T130 61V622ZM507 488Q507 514 506 528T500 564T483 597T450 620T397 635Q385 637 307 637H286Q237 637 234 628Q231 624 231 483V342H302H339Q390 342 423 349T481 382Q507 411 507 488Z" transform="translate(736,0)" style="stroke-width:3;"></path><path data-c="4B" d="M128 622Q121 629 117 631T101 634T58 637H25V683H36Q57 680 180 680Q315 680 324 683H335V637H313Q235 637 233 620Q232 618 232 462L233 307L379 449Q425 494 479 546Q518 584 524 591T531 607V608Q531 630 503 636Q501 636 498 636T493 637H489V683H499Q517 680 630 680Q704 680 716 683H722V637H708Q633 633 589 597Q584 592 495 506T406 419T515 254T631 80Q644 60 662 54T715 46H736V0H728Q719 3 615 3Q493 3 472 0H461V46H469Q515 46 515 72Q515 78 512 84L336 351Q332 348 278 296L232 251V156Q232 62 235 58Q243 47 302 46H335V0H324Q303 3 180 3Q45 3 36 0H25V46H58Q100 47 109 49T128 61V622Z" transform="translate(1417,0)" style="stroke-width:3;"></path><path data-c="4D" d="M132 622Q125 629 121 631T105 634T62 637H29V683H135Q221 683 232 682T249 675Q250 674 354 398L458 124L562 398Q666 674 668 675Q671 681 683 682T781 683H887V637H854Q814 636 803 634T785 622V61Q791 51 802 49T854 46H887V0H876Q855 3 736 3Q605 3 596 0H585V46H618Q660 47 669 49T688 61V347Q688 424 688 461T688 546T688 613L687 632Q454 14 450 7Q446 1 430 1T410 7Q409 9 292 316L176 624V606Q175 588 175 543T175 463T175 356L176 86Q187 50 261 46H278V0H269Q254 3 154 3Q52 3 37 0H29V46H46Q78 48 98 56T122 69T132 86V622Z" transform="translate(2195,0)" style="stroke-width:3;"></path></g><g data-mml-node="mo" transform="translate(3389.8,0)"><path data-c="3D" d="M56 347Q56 360 70 367H707Q722 359 722 347Q722 336 708 328L390 327H72Q56 332 56 347ZM56 153Q56 168 72 173H708Q722 163 722 153Q722 140 707 133H70Q56 140 56 153Z" style="stroke-width:3;"></path></g><g data-mml-node="mfrac" transform="translate(4445.6,0)"><g data-mml-node="TeXAtom" data-mjx-texclass="ORD" transform="translate(220,676)"><g data-mml-node="msup"><g data-mml-node="mn"><path data-c="31" d="M213 578L200 573Q186 568 160 563T102 556H83V602H102Q149 604 189 617T245 641T273 663Q275 666 285 666Q294 666 302 660V361L303 61Q310 54 315 52T339 48T401 46H427V0H416Q395 3 257 3Q121 3 100 0H88V46H114Q136 46 152 46T177 47T193 50T201 52T207 57T213 61V578Z" style="stroke-width:3;"></path><path data-c="30" d="M96 585Q152 666 249 666Q297 666 345 640T423 548Q460 465 460 320Q460 165 417 83Q397 41 362 16T301 -15T250 -22Q224 -22 198 -16T137 16T82 83Q39 165 39 320Q39 494 96 585ZM321 597Q291 629 250 629Q208 629 178 597Q153 571 145 525T137 333Q137 175 145 125T181 46Q209 16 250 16Q290 16 318 46Q347 76 354 130T362 333Q362 478 354 524T321 597Z" transform="translate(500,0)" style="stroke-width:3;"></path></g><g data-mml-node="mn" transform="translate(1033,393.1) scale(0.707)"><path data-c="39" d="M352 287Q304 211 232 211Q154 211 104 270T44 396Q42 412 42 436V444Q42 537 111 606Q171 666 243 666Q245 666 249 666T257 665H261Q273 665 286 663T323 651T370 619T413 560Q456 472 456 334Q456 194 396 97Q361 41 312 10T208 -22Q147 -22 108 7T68 93T121 149Q143 149 158 135T173 96Q173 78 164 65T148 49T135 44L131 43Q131 41 138 37T164 27T206 22H212Q272 22 313 86Q352 142 352 280V287ZM244 248Q292 248 321 297T351 430Q351 508 343 542Q341 552 337 562T323 588T293 615T246 625Q208 625 181 598Q160 576 154 546T147 441Q147 358 152 329T172 282Q197 248 244 248Z" style="stroke-width:3;"></path></g></g><g data-mml-node="mo" transform="translate(1658.8,0)"><path data-c="D7" d="M630 29Q630 9 609 9Q604 9 587 25T493 118L389 222L284 117Q178 13 175 11Q171 9 168 9Q160 9 154 15T147 29Q147 36 161 51T255 146L359 250L255 354Q174 435 161 449T147 471Q147 480 153 485T168 490Q173 490 175 489Q178 487 284 383L389 278L493 382Q570 459 587 475T609 491Q630 491 630 471Q630 464 620 453T522 355L418 250L522 145Q606 61 618 48T630 29Z" style="stroke-width:3;"></path></g><g data-mml-node="mi" transform="translate(2659,0)"><path data-c="1D436" d="M50 252Q50 367 117 473T286 641T490 704Q580 704 633 653Q642 643 648 636T656 626L657 623Q660 623 684 649Q691 655 699 663T715 679T725 690L740 705H746Q760 705 760 698Q760 694 728 561Q692 422 692 421Q690 416 687 415T669 413H653Q647 419 647 422Q647 423 648 429T650 449T651 481Q651 552 619 605T510 659Q484 659 454 652T382 628T299 572T226 479Q194 422 175 346T156 222Q156 108 232 58Q280 24 350 24Q441 24 512 92T606 240Q610 253 612 255T628 257Q648 257 648 248Q648 243 647 239Q618 132 523 55T319 -22Q206 -22 128 53T50 252Z" style="stroke-width:3;"></path></g></g><g data-mml-node="TeXAtom" data-mjx-texclass="ORD" transform="translate(533.8,-686)"><g data-mml-node="mi"><path data-c="1D441" d="M234 637Q231 637 226 637Q201 637 196 638T191 649Q191 676 202 682Q204 683 299 683Q376 683 387 683T401 677Q612 181 616 168L670 381Q723 592 723 606Q723 633 659 637Q635 637 635 648Q635 650 637 660Q641 676 643 679T653 683Q656 683 684 682T767 680Q817 680 843 681T873 682Q888 682 888 672Q888 650 880 642Q878 637 858 637Q787 633 769 597L620 7Q618 0 599 0Q585 0 582 2Q579 5 453 305L326 604L261 344Q196 88 196 79Q201 46 268 46H278Q284 41 284 38T282 19Q278 6 272 0H259Q228 2 151 2Q123 2 100 2T63 2T46 1Q31 1 31 10Q31 14 34 26T39 40Q41 46 62 46Q130 49 150 85Q154 91 221 362L289 634Q287 635 234 637Z" style="stroke-width:3;"></path></g><g data-mml-node="mo" transform="translate(1110.2,0)"><path data-c="D7" d="M630 29Q630 9 609 9Q604 9 587 25T493 118L389 222L284 117Q178 13 175 11Q171 9 168 9Q160 9 154 15T147 29Q147 36 161 51T255 146L359 250L255 354Q174 435 161 449T147 471Q147 480 153 485T168 490Q173 490 175 489Q178 487 284 383L389 278L493 382Q570 459 587 475T609 491Q630 491 630 471Q630 464 620 453T522 355L418 250L522 145Q606 61 618 48T630 29Z" style="stroke-width:3;"></path></g><g data-mml-node="mi" transform="translate(2110.4,0)"><path data-c="1D43F" d="M228 637Q194 637 192 641Q191 643 191 649Q191 673 202 682Q204 683 217 683Q271 680 344 680Q485 680 506 683H518Q524 677 524 674T522 656Q517 641 513 637H475Q406 636 394 628Q387 624 380 600T313 336Q297 271 279 198T252 88L243 52Q243 48 252 48T311 46H328Q360 46 379 47T428 54T478 72T522 106T564 161Q580 191 594 228T611 270Q616 273 628 273H641Q647 264 647 262T627 203T583 83T557 9Q555 4 553 3T537 0T494 -1Q483 -1 418 -1T294 0H116Q32 0 32 10Q32 17 34 24Q39 43 44 45Q48 46 59 46H65Q92 46 125 49Q139 52 144 61Q147 65 216 339T285 628Q285 635 228 637Z" style="stroke-width:3;"></path></g></g><rect width="3619" height="60" x="120" y="220"></rect></g></g></g>',1)]))),t[3]||(t[3]=e("mjx-assistive-mml",{unselectable:"on",display:"block",style:{top:"0px",left:"0px",clip:"rect(1px, 1px, 1px, 1px)","-webkit-touch-callout":"none","-webkit-user-select":"none","-khtml-user-select":"none","-moz-user-select":"none","-ms-user-select":"none","user-select":"none",position:"absolute",padding:"1px 0px 0px 0px",border:"0px",display:"block",overflow:"hidden",width:"100%"}},[e("math",{xmlns:"http://www.w3.org/1998/Math/MathML",display:"block"},[e("mtext",null,"RPKM"),e("mo",null,"="),e("mfrac",null,[e("mrow",{"data-mjx-texclass":"ORD"},[e("msup",null,[e("mn",null,"10"),e("mn",null,"9")]),e("mo",null,"×"),e("mi",null,"C")]),e("mrow",{"data-mjx-texclass":"ORD"},[e("mi",null,"N"),e("mo",null,"×"),e("mi",null,"L")])])])],-1))]),t[19]||(t[19]=e("p",null,"Where:",-1)),e("ul",null,[e("li",null,[e("mjx-container",Q,[(n(),i("svg",T,t[4]||(t[4]=[e("g",{stroke:"currentColor",fill:"currentColor","stroke-width":"0",transform:"scale(1,-1)"},[e("g",{"data-mml-node":"math"},[e("g",{"data-mml-node":"mi"},[e("path",{"data-c":"1D436",d:"M50 252Q50 367 117 473T286 641T490 704Q580 704 633 653Q642 643 648 636T656 626L657 623Q660 623 684 649Q691 655 699 663T715 679T725 690L740 705H746Q760 705 760 698Q760 694 728 561Q692 422 692 421Q690 416 687 415T669 413H653Q647 419 647 422Q647 423 648 429T650 449T651 481Q651 552 619 605T510 659Q484 659 454 652T382 628T299 572T226 479Q194 422 175 346T156 222Q156 108 232 58Q280 24 350 24Q441 24 512 92T606 240Q610 253 612 255T628 257Q648 257 648 248Q648 243 647 239Q618 132 523 55T319 -22Q206 -22 128 53T50 252Z",style:{"stroke-width":"3"}})])])],-1)]))),t[5]||(t[5]=e("mjx-assistive-mml",{unselectable:"on",display:"inline",style:{top:"0px",left:"0px",clip:"rect(1px, 1px, 1px, 1px)","-webkit-touch-callout":"none","-webkit-user-select":"none","-khtml-user-select":"none","-moz-user-select":"none","-ms-user-select":"none","user-select":"none",position:"absolute",padding:"1px 0px 0px 0px",border:"0px",display:"block",width:"auto",overflow:"hidden"}},[e("math",{xmlns:"http://www.w3.org/1998/Math/MathML"},[e("mi",null,"C")])],-1))]),t[6]||(t[6]=a(" = the number of mapped reads for the specified transcript."))]),e("li",null,[e("mjx-container",f,[(n(),i("svg",b,t[7]||(t[7]=[e("g",{stroke:"currentColor",fill:"currentColor","stroke-width":"0",transform:"scale(1,-1)"},[e("g",{"data-mml-node":"math"},[e("g",{"data-mml-node":"mi"},[e("path",{"data-c":"1D441",d:"M234 637Q231 637 226 637Q201 637 196 638T191 649Q191 676 202 682Q204 683 299 683Q376 683 387 683T401 677Q612 181 616 168L670 381Q723 592 723 606Q723 633 659 637Q635 637 635 648Q635 650 637 660Q641 676 643 679T653 683Q656 683 684 682T767 680Q817 680 843 681T873 682Q888 682 888 672Q888 650 880 642Q878 637 858 637Q787 633 769 597L620 7Q618 0 599 0Q585 0 582 2Q579 5 453 305L326 604L261 344Q196 88 196 79Q201 46 268 46H278Q284 41 284 38T282 19Q278 6 272 0H259Q228 2 151 2Q123 2 100 2T63 2T46 1Q31 1 31 10Q31 14 34 26T39 40Q41 46 62 46Q130 49 150 85Q154 91 221 362L289 634Q287 635 234 637Z",style:{"stroke-width":"3"}})])])],-1)]))),t[8]||(t[8]=e("mjx-assistive-mml",{unselectable:"on",display:"inline",style:{top:"0px",left:"0px",clip:"rect(1px, 1px, 1px, 1px)","-webkit-touch-callout":"none","-webkit-user-select":"none","-khtml-user-select":"none","-moz-user-select":"none","-ms-user-select":"none","user-select":"none",position:"absolute",padding:"1px 0px 0px 0px",border:"0px",display:"block",width:"auto",overflow:"hidden"}},[e("math",{xmlns:"http://www.w3.org/1998/Math/MathML"},[e("mi",null,"N")])],-1))]),t[9]||(t[9]=a(" = the total number of mapped reads in the experiment."))]),e("li",null,[e("mjx-container",A,[(n(),i("svg",x,t[10]||(t[10]=[e("g",{stroke:"currentColor",fill:"currentColor","stroke-width":"0",transform:"scale(1,-1)"},[e("g",{"data-mml-node":"math"},[e("g",{"data-mml-node":"mi"},[e("path",{"data-c":"1D43F",d:"M228 637Q194 637 192 641Q191 643 191 649Q191 673 202 682Q204 683 217 683Q271 680 344 680Q485 680 506 683H518Q524 677 524 674T522 656Q517 641 513 637H475Q406 636 394 628Q387 624 380 600T313 336Q297 271 279 198T252 88L243 52Q243 48 252 48T311 46H328Q360 46 379 47T428 54T478 72T522 106T564 161Q580 191 594 228T611 270Q616 273 628 273H641Q647 264 647 262T627 203T583 83T557 9Q555 4 553 3T537 0T494 -1Q483 -1 418 -1T294 0H116Q32 0 32 10Q32 17 34 24Q39 43 44 45Q48 46 59 46H65Q92 46 125 49Q139 52 144 61Q147 65 216 339T285 628Q285 635 228 637Z",style:{"stroke-width":"3"}})])])],-1)]))),t[11]||(t[11]=e("mjx-assistive-mml",{unselectable:"on",display:"inline",style:{top:"0px",left:"0px",clip:"rect(1px, 1px, 1px, 1px)","-webkit-touch-callout":"none","-webkit-user-select":"none","-khtml-user-select":"none","-moz-user-select":"none","-ms-user-select":"none","user-select":"none",position:"absolute",padding:"1px 0px 0px 0px",border:"0px",display:"block",width:"auto",overflow:"hidden"}},[e("math",{xmlns:"http://www.w3.org/1998/Math/MathML"},[e("mi",null,"L")])],-1))]),t[12]||(t[12]=a(" = the length of the specified transcript in base pairs."))])]),t[20]||(t[20]=r('<h3 id="genomic-data-mapping-and-analysis-workflow" tabindex="-1">Genomic Data Mapping and Analysis Workflow <a class="header-anchor" href="#genomic-data-mapping-and-analysis-workflow" aria-label="Permalink to &quot;Genomic Data Mapping and Analysis Workflow&quot;">​</a></h3><ul><li><p><strong>Reads Mapping</strong>: The colorful fragments on the left represent reads obtained from sequencing data. Through a process called <strong>mapping</strong>, these reads are aligned to the reference genome (shown as a DNA structure on the right) to determine their positions within the genome.</p></li><li><p><strong>Mapped Alignment</strong>: Once mapped, the reads are aligned to the reference genome, forming a <strong>mapped alignment</strong> (illustrated as green bars). This step accurately locates the origin of each read within the genome.</p></li><li><p><strong>Analysis Applications</strong>:</p><ul><li><strong>Calling Genetic Variants</strong>: By comparing the sequencing data with the reference genome, genetic variants, such as single nucleotide polymorphisms (<em>SNPs</em>), can be identified.</li><li><strong>Measuring Abundance</strong>: Mapped data can be used for abundance analysis, such as RNA-Seq to measure gene expression levels, and ChIP-Seq to analyze protein-DNA interactions. These analyses help in understanding gene expression and regulatory patterns under various conditions.</li></ul></li></ul><h4 id="mapping-reads-from-rna-seq" tabindex="-1">Mapping Reads from RNA-Seq <a class="header-anchor" href="#mapping-reads-from-rna-seq" aria-label="Permalink to &quot;Mapping Reads from RNA-Seq&quot;">​</a></h4><ul><li><strong>Mapping RNA-Seq Reads</strong>: RNA-Seq reads are mapped to the reference genome to locate where each read aligns.</li><li><strong>Junction Sites</strong>: Some reads span exon-exon junctions, indicating splice sites where introns have been removed during RNA processing.</li><li><strong>Detection of Novel Isoforms</strong>: By analyzing these <strong>junction reads</strong>, novel splicing isoforms can be identified, revealing alternative splicing patterns and adding insight into gene expression diversity.</li></ul><h2 id="chapter-2-part3-alignment-and-ngs-reads-mapping-blast" tabindex="-1">Chapter 2 Part3- Alignment and NGS reads Mapping - BLAST <a class="header-anchor" href="#chapter-2-part3-alignment-and-ngs-reads-mapping-blast" aria-label="Permalink to &quot;Chapter 2 Part3- Alignment and NGS reads Mapping - BLAST&quot;">​</a></h2><h2 id="chapter-3-part1-gene-ontology" tabindex="-1">Chapter 3 Part1- Gene Ontology <a class="header-anchor" href="#chapter-3-part1-gene-ontology" aria-label="Permalink to &quot;Chapter 3 Part1- Gene Ontology&quot;">​</a></h2><ul><li>Molecular Function <em>elemental activity/task</em></li><li>Biological Process <em>biological goal or objective</em></li><li>Cellular Component <em>location or complex</em></li></ul><h3 id="go-relationships" tabindex="-1">GO Relationships <a class="header-anchor" href="#go-relationships" aria-label="Permalink to &quot;GO Relationships&quot;">​</a></h3><ul><li><em>IS A</em></li><li><em>PART OF</em></li><li><em>REGULATES</em><ul><li><em>POSITIVELY REGULATES</em></li><li><em>NEGATIVELY REGULATES</em></li></ul></li></ul><p>这几种格式（<strong>FASTA</strong>、<strong>FASTQ</strong>、<strong>SAM</strong>、<strong>VCF</strong>、<strong>GFF</strong>、<strong>PDB</strong>）都是生物信息学中常用的数据文件格式，各自有不同的用途和结构。以下是每种格式的总结：</p><h3 id="_1-fasta" tabindex="-1">1. FASTA <a class="header-anchor" href="#_1-fasta" aria-label="Permalink to &quot;1. FASTA&quot;">​</a></h3><ul><li><strong>用途</strong>：存储<strong>序列数据</strong>（如DNA、RNA或蛋白质序列）。</li><li><strong>结构</strong>： <ul><li>每个序列由两行组成：第一行以<code>&gt;</code>开头，包含序列的ID或描述；第二行是实际的序列。</li></ul></li><li><strong>特点</strong>：只包含序列信息，没有质量分数。适合基因组或蛋白质序列的基本存储和共享。</li></ul><h3 id="_2-fastq" tabindex="-1">2. FASTQ <a class="header-anchor" href="#_2-fastq" aria-label="Permalink to &quot;2. FASTQ&quot;">​</a></h3><ul><li><strong>用途</strong>：存储<strong>高通量测序数据</strong>，包含序列和质量信息。</li><li><strong>结构</strong>： <ul><li>每个条目由四行组成：第一行以<code>@</code>开头，表示序列标识符；第二行是序列；第三行以<code>+</code>开头，标记质量分数的开始；第四行是质量信息，与序列长度相同。</li></ul></li><li><strong>特点</strong>：包含每个碱基的测序质量分数，常用于测序数据分析，便于质量控制。</li></ul><h3 id="_3-sam-sequence-alignment-map" tabindex="-1">3. SAM (Sequence Alignment/Map) <a class="header-anchor" href="#_3-sam-sequence-alignment-map" aria-label="Permalink to &quot;3. SAM (Sequence Alignment/Map)&quot;">​</a></h3><ul><li><strong>用途</strong>：存储<strong>序列比对信息</strong>，记录reads如何映射到参考基因组上。</li><li><strong>结构</strong>： <ul><li>文本格式，每行记录一个比对结果，包括序列名称、比对位置、匹配质量等信息。</li></ul></li><li><strong>特点</strong>：SAM格式文件大，但信息全面。常用于分析比对结果，可以转换成更紧凑的二进制格式（BAM）。</li></ul><h3 id="_4-vcf-variant-call-format" tabindex="-1">4. VCF (Variant Call Format) <a class="header-anchor" href="#_4-vcf-variant-call-format" aria-label="Permalink to &quot;4. VCF (Variant Call Format)&quot;">​</a></h3><ul><li><strong>用途</strong>：存储<strong>遗传变异信息</strong>，如SNPs和Indels。</li><li><strong>结构</strong>： <ul><li>文本格式，包含头信息和每个变异的记录，每行包含染色体位置、参考碱基、变异碱基、质量分数等信息。</li></ul></li><li><strong>特点</strong>：标准化格式，适合基因组变异数据的存储和共享，常用于基因组学和变异分析。</li></ul><h3 id="_5-gff-general-feature-format-gtf-general-transfer-format" tabindex="-1">5. GFF (General Feature Format) / GTF (General Transfer Format) <a class="header-anchor" href="#_5-gff-general-feature-format-gtf-general-transfer-format" aria-label="Permalink to &quot;5. GFF (General Feature Format) / GTF (General Transfer Format)&quot;">​</a></h3><ul><li><strong>用途</strong>：存储<strong>基因组注释信息</strong>，描述基因组上各种功能元件的位置，如基因、外显子、启动子等。</li><li><strong>结构</strong>： <ul><li>每行记录一个特征，包括染色体号、起始和结束位置、特征类型（如基因、外显子）、方向、附加信息等。</li></ul></li><li><strong>特点</strong>：标准化格式，常用于描述基因组的结构和功能注释。</li></ul><h3 id="_6-pdb-protein-data-bank" tabindex="-1">6. PDB (Protein Data Bank) <a class="header-anchor" href="#_6-pdb-protein-data-bank" aria-label="Permalink to &quot;6. PDB (Protein Data Bank)&quot;">​</a></h3><ul><li><strong>用途</strong>：存储<strong>蛋白质三维结构</strong>信息，通常用于结构生物学。</li><li><strong>结构</strong>： <ul><li>文本格式，记录每个原子的位置、原子类型、氨基酸序列等信息。</li></ul></li><li><strong>特点</strong>：常用于可视化蛋白质的三维结构，适合蛋白质功能和相互作用研究。</li></ul><h3 id="总结表格" tabindex="-1">总结表格 <a class="header-anchor" href="#总结表格" aria-label="Permalink to &quot;总结表格&quot;">​</a></h3><table tabindex="0"><thead><tr><th>格式</th><th>用途</th><th>包含信息</th><th>特点</th></tr></thead><tbody><tr><td>FASTA</td><td>序列存储</td><td>序列</td><td>结构简单，适合基本序列存储</td></tr><tr><td>FASTQ</td><td>高通量测序数据</td><td>序列 + 质量信息</td><td>包含质量分数，适合测序数据分析</td></tr><tr><td>SAM</td><td>比对信息</td><td>序列与参考基因组的比对位置</td><td>信息全面，文件较大</td></tr><tr><td>VCF</td><td>遗传变异信息</td><td>变异位点和变异信息</td><td>标准格式，用于变异分析</td></tr><tr><td>GFF</td><td>基因组注释</td><td>基因、外显子等功能元件位置</td><td>常用于基因组功能注释</td></tr><tr><td>PDB</td><td>蛋白质三维结构</td><td>原子坐标和氨基酸序列</td><td>用于蛋白质结构分析和可视化</td></tr></tbody></table><p>这几种格式各有用途，分别适用于序列数据存储、测序数据质量分析、比对分析、基因组变异研究、基因组注释和蛋白质结构研究等领域。</p>',25))])}const D=g(p,[["render",w]]);export{H as __pageData,D as default};
